code/modules/ghostroles/spawner/simplemob/maintdrone.dm

```diff
@@ -1,20 +1,30 @@
 /datum/ghostspawner/simplemob/maintdrone
 	short_name = "maintdrone"
-	name = "Maintenence Drone"
-	desc = "Maintain and Improve the Systems on the Aurora"
+	name = "Maintenance Drone"
+	desc = "Maintain and Improve the Systems on the Aurora."
 	show_on_job_select = FALSE
 	tags = list("Simple Mobs")
 
 	respawn_flag = MINISYNTH //Flag to check for when trying to spawn someone of that type (CREW, ANIMAL, MINISYNTH)
 	jobban_job = "Cyborg"
-	loc_type = GS_LOC_ATOM
 
 	//Vars regarding the mob to use
-	spawn_mob = /mob/living/simple_animal/rat //The mob that should be spawned
+	spawn_mob = /mob/living/silicon/robot/drone //The mob that should be spawned
 
-/datum/ghostspawner/simplemob/maintdrone/cant_see()
+// we fake it here to ensure it pops up and gets added to SSghostroles.spawners, handling of the fabricators is done below
+/datum/ghostspawner/simplemob/maintdrone/select_spawnlocation()
+	return TRUE
```

Comment:
Ghostspawner initialization is checked after `SS_INIT_ATOMS` and `SS_INIT_MACHINERY`. I think this ghostspawner should be able to check if there are any fabricators mapped at the start -- You can then make creation or destruction of a fabricator trigger an update for this ghostspawner.

Reply:
ghost roles run this check at roundstart to see if they get added to the SSghostrole subsystem. if this check fails the first time, readding it to the subsystem requires an even bigger hack than this one, which is why i opted for this